@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -313,46 +345,118 @@ export function placePointLabel(
-        tmpMeasurementParams.path = undefined;
-        tmpMeasurementParams.pathOverflow = false;
-        tmpMeasurementParams.letterCaseArray = label.glyphCaseArray!;
-        textCanvas.measureText(label.glyphs!, label.bounds, tmpMeasurementParams);
     }
 
-    screenPosition.add(computePointTextOffset(label, scale, tmpTextOffset));
-    outScreenPosition.set(screenPosition.x, screenPosition.y, labelState.renderDistance);
-
-    // TODO: Make the margin configurable
-    tmpBox.copy(label.bounds!).expandByVector(pointLabelMargin);
-    tmpBox.min.multiplyScalar(scale);
-    tmpBox.max.multiplyScalar(scale);
-    tmpBox.translate(screenPosition);
-    tmp2DBox.set(
-        tmpBox.min.x,
-        tmpBox.min.y,
-        tmpBox.max.x - tmpBox.min.x,
-        tmpBox.max.y - tmpBox.min.y
-    );
-
-    // Check the text visibility.
-    if (!screenCollisions.isVisible(tmp2DBox)) {
-        return PlacementResult.Invisible;
-    }
+    // Setup measurements parameters for textCanvas.measureText().
+    tmpMeasurementParams.outputCharacterBounds = undefined;
+    tmpMeasurementParams.path = undefined;
+    tmpMeasurementParams.pathOverflow = false;
+    tmpMeasurementParams.letterCaseArray = label.glyphCaseArray!;
+
+    // NOTE:
+    // We may save some CPU time by firstly checking visibility of all possible text placements,
+    // their bounding boxes merged altogether, this may save time for further calculations.
+    // Would be enough to include TL and BR placements.
+
+    // Start with currently set alignment settings.
+    const basePlacement = {
+        h: label.layoutStyle!.horizontalAlignment,
+        v: label.layoutStyle!.verticalAlignment
+    };
+    // Placements options will be read from label.layoutStyle.placements in final solution.
+    const placements = possibleAnchorPlacementsCW;
+    // Find first anchor placement.
+    const idx = placements.findIndex(val => {
+        return val.h === basePlacement.h && val.v === basePlacement.v;
+    });
+    // If not listed in placement options list (idx === -1) will start from current alignment setup.
+    const placementsNum = placements.length;
+    let textOffset: THREE.Vector2;
+    // Will be true if all text placements are invisible.
+    let allInvisible: boolean = true;
+    let i = 0;
+    for (i = 0; i < placementsNum; ++i) {
+        const anchorPlacement = idx + i < 0 ? basePlacement : placements[(idx + i) % placementsNum];
+        // Override label text alignment for measurements and leave it so if passed collisions test.
+        // NOTE: This actually writes to label.layoutStyle.verticalAlignment/horizontalAlignment.
+        textCanvas.textLayoutStyle.horizontalAlignment = anchorPlacement.h;
+        textCanvas.textLayoutStyle.verticalAlignment = anchorPlacement.v;
+
+        // Compute label bounds according to layout settings.
+        textCanvas.measureText(label.glyphs!, label.bounds, tmpMeasurementParams);
```

Comment:
Why are you computing the label bounds for every possible placement? They are independent from the label position right? 
They could be computed once and cached in textElement just as it was done before.